source/core/slang-io.cpp

```diff
@@ -40,15 +40,15 @@ namespace Slang
 #endif
 	}
 
-	String Path::TruncateExt(const String & path)
+	String Path::truncateExt(const String& path)
```

Comment:
Not really relevant to this PR (since this is more about naming convention than the specific choices), but I'd personally go for `Extension` over `Ext` for these functions just for clarity. There's just too much precedent with stuff like `CreateWindowEx` that uses `Ex`/`Ext` as a suffix for an "extended" version of a function for the intent here to be 100% clear, even with the context that this is for manipulating paths.

Another naming nit here is that the extension isn't really be *truncated* but rather removed entirely. Something like `trimExtension` or `removeExtension` might be an altogether better name.

And then there's the fact that given a path like `/foo/bar.zip/something` the current implementation would split it down to `/foo/bar` which isn't correct.

Anyway, like I said at the start of the comment: none of this is really about the current PR.